api/include/opentelemetry/trace/tracer-registry.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "./tracer.h"
+
+namespace opentelemetry
+{
+namespace trace
+{
+class Tracer;
+class TracerRegistry
```

Comment:
My understanding of the current specification is that there is not a Registry requirement. The only requirement is that the user provide a library name (and optional version) in order to obtain a Tracer.

The Java implementation has, for forward-looking reasons, chosen to keep a map of these, but there is no reason they need to be "de-duplicated" by map in the specification.

Reply:
I'll throw them into a different container in that case, thanks!

For my own edification, it sounds like there could potentially be two tracers being used for the same library/version pair with different configurations? I was under the impression that the library details would fully specify the desired configuration.

Reply:
This has been a confusing feature addition ("Named tracers"). My understanding is that there is a common implementation (called it the "SDK" for lack of a better term) that is shared by all the Tracers returned by the factory/provider API, so they should have the same configuration.

If you think of Tracer as just a 3-tuple of { name, version, *SDK }, then it should be clear there's not much reason to implement a registry. It's only when you begin having heavy-weight optimizations based on the name/version that you might consider a map to maintain uniqueness.

Reply:
Awesome, thanks for the clarification!